envoy-config.yaml

```
@@ -1,63 +1,62 @@
-                codec_type: auto
-                stat_prefix: ingress_http
-                route_config:
-                  name: local_route
-                  virtual_hosts:
-                    - name: local_service
-                      domains:
-                        - "*"
-                      routes:
-                        - match:
-                            prefix: "/"
-                          route:
-                            cluster: web_service
-
-                http_filters:
-                  - name: envoy.filters.http.wasm
-                    typed_config:
-                      "@type": type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
-                      config:
-                        name: "my_plugin"
-                        root_id: "my_root_id"
-                        configuration:
-                          "@type": "type.googleapis.com/google.protobuf.StringValue"
-                          value: |
-                            SecDebugLogLevel 5
-                            SecDebugLog modsec.log
-                            SecRuleEngine On
-                            SecRule REQUEST_URI "@streq /admin" "id:1,phase:1,t:lowercase,deny"
-                        vm_config:
-                          runtime: "envoy.wasm.runtime.v8"
-                          vm_id: "my_vm_id"
-                          configuration: {}
-                          code:
-                            local:
-                              filename: ./bin/modsecurity-filter.wasm
-                  - name: envoy.filters.http.router
-                    typed_config: {}
+  - name: listener_name
+    address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 8001
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          stat_prefix: ingress_http
+          codec_type: auto
+          route_config:
+            name: local_route
+            virtual_hosts:
+              - name: local_service
+                domains:
+                  - "*"
+                routes:
+                  - match:
+                      prefix: "/"
+                    route:
+                      cluster: web_service
+          http_filters:
+          - name: envoy.filters.http.wasm
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
+              config:
+                name: "my_plugin"
+                root_id: ""
+                configuration:
+                  "@type": "type.googleapis.com/google.protobuf.StringValue"
+                  value: |
+                    {
+                      "rules":"SecDebugLogLevel 5\nSecDebugLog modsec.log\nSecRuleEngine On\nSecRule REQUEST_URI \"@streqr /admin\" \"id:101,phase:1,t:lowercase,deny\""
```

```suggestion
                      "rules":"
                         SecDebugLogLevel 5
                         SecDebugLog modsec.log
                         SecRuleEngine On\nSecRule REQUEST_URI \"@streqr /admin\" \"id:101,phase:1,t:lowercase,deny\"
                      "
```
Would something like this also work?